### ISSUE TITLE
Unable to login with certificate only

### ISSUE THREAD

**Issue by kamushadenes** [adrienverge/openfortivpn#602]
Hello people,

I have seen https://github.com/adrienverge/openfortivpn/issues/375 which led to https://github.com/adrienverge/openfortivpn/pull/493 but I'm still unable to login to my cert-only work VPN. Version is 1.11.0 from Arch `community` repository (`community/openfortivpn 1.11.0-1`)

Connecting with the official client works as expected, the only difference is that the certificate is a p12 while with openfortivpn I converted it to PEM.

```
❯ sudo openfortivpn -v HOST:10443 --user-cert=network_cert.pem --user-key=network_key.pem  --trusted-cert XXXXXXXXXXXXXXXXXXXXX --password="" --username="CN_FIELD_ON_THE_CERTIFICATE" --ca-file=network_ca.pem
DEBUG:  openfortivpn 1.11.0
WARN:   Bad port in config file: "0".
DEBUG:  Loaded config file "/etc/openfortivpn/config".
DEBUG:  Config host = "HOST"
DEBUG:  Config realm = ""
DEBUG:  Config port = "10443"
DEBUG:  Config username = "CN_FIELD_ON_THE_CERTIFICATE"
DEBUG:  Resolving gateway host ip
DEBUG:  Establishing ssl connection
DEBUG:  server_addr: XX.XX.XX.XX
DEBUG:  server_port: 10443
DEBUG:  gateway_addr: XX.XX.XX.XX
DEBUG:  gateway_port: 10443
DEBUG:  Setting cipher list to: HIGH:!aNULL:!kRSA:!PSK:!SRP:!MD5:!RC4DEBUG:  Gateway certificate validation succeeded.
INFO:   Connected to gateway.
ERROR:  Could not authenticate to gateway. Please check the password, client certificate, etc.
DEBUG:  HTTP status code 405
INFO:   Closed connection to gateway.
DEBUG:  server_addr: XX.XX.XX.XX
DEBUG:  server_port: 10443
DEBUG:  gateway_addr: XX.XX.XX.XX
DEBUG:  gateway_port: 10443
DEBUG:  Setting cipher list to: HIGH:!aNULL:!kRSA:!PSK:!SRP:!MD5:!RC4DEBUG:  Gateway certificate validation succeeded.
INFO:   Logged out.
```

I also tried a dummy password instead of a blank one and several different username variants.

For example: 

```
CN field: net-firstname.lastname@domain.com

Username: net-firstname.lastname@domain.com
Username firstname.lastname@domain.com
Username: net-firstname.lastname
Username: firstname.lastname
```

But to no avail.

Any ideas?

Thanks in advance.

**Comment by DimitriPapadopoulos** [comment]
> Connecting with the official client works as expected, the only difference is that the certificate is a p12 while with openfortivpn I converted it to PEM.

Just to make sure, on which platform does the official client works? Windows or Linux? I'm asking because on Windows _FortiClient_ is often configured to use IPSec while on Linux it is limited to SSL.

**Comment by kamushadenes** [comment]
Linux, the [AUR](https://aur.archlinux.org/packages/forticlientsslvpn/) one

**Comment by mrbaseman** [comment]
try with empty user name and empty password  `sudo openfortivpn  HOST:PORT --username="" --password="" ...`

**Comment by kamushadenes** [comment]
DEBUG:  openfortivpn 1.11.0
WARN:   Bad port in config file: "0".
DEBUG:  Loaded config file "/etc/openfortivpn/config".
ERROR:  Specify an username.

**Comment by DimitriPapadopoulos** [comment]
@kamushadenes Can you (temporarily) get `/etc/openfortivpn/config` out of the way, again just to make sure?
```
sudo mv /etc/openfortivpn/config /etc/openfortivpn/config.bak
```

**Comment by mrbaseman** [comment]
hmm... and the  `ERROR: Specify an username.` should not appear since 1.11.0 if a user certificate (and key) is supplied (sorry, this was what I wanted to denote with the dots `...` in my [post above](https://github.com/adrienverge/openfortivpn/issues/602#issuecomment-604508959))

**Comment by kamushadenes** [comment]
@DimitriPapadopoulos same behaviour

**Comment by kamushadenes** [comment]
I compiled ' 1.13.2' using the [official PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/openfortivpn) for '1.11.0' and it worked, guess it's a bug on the [release](https://github.com/adrienverge/openfortivpn/releases/tag/v1.11.0)?

**Comment by kamushadenes** [comment]
Now I need to convince network manager to accept an empty username on it's plugin.

**Comment by kamushadenes** [comment]
Tracking NM bug at https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn/-/issues/23

**Comment by mrbaseman** [comment]
> I compiled `1.13.2` using the [official PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/openfortivpn) for `1.11.0` and it worked, guess it's a bug on the [release](https://github.com/adrienverge/openfortivpn/releases/tag/v1.11.0)?

Ah, this could be #496 which was fixed in 1.12.0

**Comment by kamushadenes** [comment]
Can confirm it works with `1.12.0`, closing this issue. 

Thank you for your time.
